### PR TITLE
navbar + menu + mobile menu

### PR DESCRIPTION
**Pull request by pananimals**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -40,6 +40,7 @@
   },
   "dependencies": {
     "bootstrap": "4.1.x",
-    "jquery": "3.3.x"
+    "jquery": "3.3.x",
+    "popper.js": "1.14.x"
```
Comment (michalkowol) [inline package.json]: Webpack.config.js - please remove:

```js
{
        from: path.resolve(__dirname, 'src/files'),
        to: path.resolve(__dirname, 'dist/files')
      }
```

it causes that problem: `ERROR in [copy-webpack-plugin] unable to locate '/Users/michal/dev/www/codeloop-website/src/files' at '/Users/michal/dev/www/codeloop-website/src/files'
C`

- [ ] src/files

File: src/css/styles/_c-btn.scss [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+  font-weight: bold;
+  color: $col-bg;
+  text-align: center;
+  text-decoration: none;
+  text-transform: uppercase;
+  letter-spacing: 0.1em;
+  background-color: $col-blue;
+  border: 0;
+  border-radius: 2rem;
+  transition-duration: 0.3s ease-in-out;
```
Comment (michalkowol) [inline src/css/styles/_c-btn.scss]: IMHO we should split `transition-duration` to:

```css
  transition-timing-function: ease-in-out;
  transition-duration: 0.3s;
```

- [ ] transition-timing-function

Ref:
* https://www.w3schools.com/csSref/css3_pr_transition-duration.asp
* https://developer.mozilla.org/en-US/docs/Web/CSS/transition-duration

File: src/css/styles/_c-btn.scss [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+    background-color: $col-green;
+
+    &:hover {
+      background-color: $col-dark-green;
+    }
+  }
+
+  &--sm {
+    padding: 0.5rem 1rem !important;
+    font-size: 0.8rem !important;
```
Comment (michalkowol) [inline src/css/styles/_c-btn.scss]: We do we need `!important`?

File: src/css/styles/_c-btn.scss (hunk 2 above)
Comment (pananimals) [reply src/css/styles/_c-btn.scss]: Is there a reason not to use a shorthand like this?

https://developer.mozilla.org/en-US/docs/Web/CSS/transition

File: src/css/styles/_c-btn.scss (hunk 3 above)
Comment (pananimals) [reply src/css/styles/_c-btn.scss]: Yes, if we hold on to idea of media-queries at the end of the file and BEM guidelines.
As `c-btn--sm` is only a modifier (ergo can't exist alone), the further media-query styling for `c-btn` overwrites its properties.

File: src/css/styles/_c-btn.scss (hunk 2 above)
Comment (michalkowol) [reply src/css/styles/_c-btn.scss]: We could use shorthand for `transition` but not for `transition-duration`

**Pull request merged**